test/suite/lib.rs

```diff
@@ -369,6 +369,73 @@ pub async fn test_tx_deduplication(ledger_env: impl LedgerEnv) -> anyhow::Result
     Ok(Outcome::Passed)
 }
 
+pub async fn test_bad_fee(ledger_env: impl LedgerEnv) -> anyhow::Result<Outcome> {
+    // Create two test accounts and transfer some tokens to the first account
+    let p1_env = setup_test_account(&ledger_env, 200_000.into()).await?;
+    let p2_env = p1_env.fork();
+
+    let mut transfer_args = Transfer::amount_to(10_000, p2_env.principal());
+    let ledger_fee = transfer_fee(&ledger_env).await.unwrap();
+    // Set incorrect fee
+    transfer_args = transfer_args.fee(ledger_fee.clone() + Nat::from(1));
+    match transfer(&ledger_env, transfer_args.clone()).await.unwrap() {
+        Ok(_) => return Err(anyhow::Error::msg("Expected Bad Fee Error")),
+        Err(err) => match err {
+            TransferError::BadFee { expected_fee } => {
+                if expected_fee != transfer_fee(&ledger_env).await.unwrap() {
+                    return Err(anyhow::Error::msg(format!(
+                        "Expected BadFee argument to be {}, got {}",
+                        ledger_fee, expected_fee
+                    )));
+                }
+            }
+            _ => return Err(anyhow::Error::msg("Expected BadFee error")),
+        },
+    }
+    Ok(Outcome::Passed)
+}
+
+pub async fn test_future_transfer(ledger_env: impl LedgerEnv) -> anyhow::Result<Outcome> {
+    // Create two test accounts and transfer some tokens to the first account
+    let p1_env = setup_test_account(&ledger_env, 200_000.into()).await?;
+    let p2_env = p1_env.fork();
+
+    let mut transfer_args = Transfer::amount_to(10_000, p2_env.principal());
+
+    // Set created time in the future
+    transfer_args = transfer_args.created_at_time(u64::MAX);
+    match transfer(&ledger_env, transfer_args)
+        .await
+        .unwrap()
+        .unwrap_err()
+    {
+        TransferError::CreatedInFuture { ledger_time: _ } => (),
```

Comment:
Nitpick: put `Ok(Outcome::Passed)` here instead of `()` and remove `return` from the `Ok` branch.